app/models/grouping.rb

```diff
@@ -105,10 +105,22 @@ class Grouping < ApplicationRecord
   has_many :starter_file_entries, through: :grouping_starter_file_entries
 
   # Assigns a random TA from a list of TAs specified by +ta_ids+ to each
-  # grouping in a list of groupings specified by +grouping_ids+. The groupings
+  # grouping in a list of groupings specified by +grouping_ids+.
+  # The ratio of groupings to ta is specified in +weightings_arr+
+  # which is a parallel array to +ta_ids+. The groupings
   # must belong to the given assignment +assignment+.
-  def self.randomly_assign_tas(grouping_ids, ta_ids, assignment)
-    assign_tas(grouping_ids, ta_ids, assignment) do |grouping_ids_, ta_ids_|
+  def self.randomly_assign_tas(grouping_ids, ta_ids, weightings_arr, assignment)
+    # Create a hash of TA's to the number of groups they are supposed to mark
+    total = weightings_arr.reduce(:+)
```

Comment:
Oh I must have missed this earlier, but you can just do `.sum` instead of `.reduce(:+)`